jpa/org.eclipse.persistence.jpa.jpql/src/main/java/org/eclipse/persistence/jpa/jpql/parser/AbstractExpression.java

```diff
@@ -525,7 +525,7 @@ public final JPQLExpression getRoot() {
      * @return  Parent expression
      */
     public final ParentExpression getParentExpression() {
-        if (this instanceof ParentExpression parentExpression) {
+        if (!(this instanceof SubExpression) && this instanceof ParentExpression parentExpression) {
```

Comment:
it would be better to add `isParentExpression: boolean` or similar to avoid casts/instanceof checks, what do you think?